base/gmp.jl

```diff
@@ -41,23 +41,30 @@ type BigInt <: Integer
     alloc::Cint
     size::Cint
     d::Ptr{Limb}
-    function BigInt()
-        b = new(zero(Cint), zero(Cint), C_NULL)
-        ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
-        finalizer(b, _gmp_clear_func)
-        return b
+    function BigInt(::Val{:allocbits}, nbits)
```

Comment:
I feared that a keyword would slow things down... the only point of this feature is to improve efficiency by avoiding allocating unnecessarily one `Limb` (the default of `mpz_init`) and then re-allocating what is needed from within `mpz_*` functions. For the same reason, I think `sizehint!` would not work. Also this is really not meant to be a user API, meant only for the wrapping layer of `mpz`, so the ugliness was kind of "on purpose". Would you accept a solution like `BigInt(::Type{Bool}, nbits)`, or maybe using an ad-hoc type tag `BigInt(::Type{SizeHint}, nbits)` ? Otherwise, I will update to use a keyword argument.